gui/layers/_markers.py

```diff
@@ -0,0 +1,279 @@
+             'If size is a list/array, must be the same length as '\
+             'coords'
+
+            if isinstance(size, list):
+                self._size = np.asarray(size)
+
+            else:
+                self._size = size
+
+            self.refresh()
+
+        else:
+            raise TypeError('size should be float or ndarray')
+
+    @property
+    def edge_width(self) -> Union[None, int, float]:
+        """None, int, float, None: width of the symbol edge in px
+        """
+
+        return self._edge_width
+
+    @edge_width.setter
+    def edge_width(self, edge_width: Union[None, float]) -> None:
+        self._edge_width = edge_width
+
+        self.refresh()
+
+    @property
+    def edge_width_rel(self) -> Union[None, int, float]:
+        """None, int, float: width of the marker edge as a fraction
+            of the marker size.
+
+            vispy docs say: "exactly one edge_width and
+            edge_width_rel must be supplied", but I don't know
+            what that means... -KY
+        """
+
+        return self._edge_width_rel
+
+    @edge_width_rel.setter
+    def edge_width_rel(self, edge_width_rel: Union[None, float]) -> None:
+        self._edge_width_rel = edge_width_rel
+
+        self.refresh()
+
+    @property
+    def edge_color(self) -> str:
+        """Color, ColorArray: the marker edge color
+        """
+
+        return self._edge_color
+
+    @edge_color.setter
+    def edge_color(self, edge_color: str) -> None:
+        self._edge_color = edge_color
+
+        self.refresh()
+
+    @property
+    def face_color(self) -> str:
+        """Color, ColorArray: color of the body of the marker
+        """
+
+        return self._face_color
+
+    @face_color.setter
+    def face_color(self, face_color: str) -> None:
+        self._face_color = face_color
+
+        self.refresh()
+
+    @property
+    def scaling(self) -> bool:
+        """bool: if True, marker rescales when zooming
+        """
+
+        return self._scaling
+
+    @scaling.setter
+    def scaling(self, scaling: bool) -> None:
+        self._scaling = scaling
+
+        self.refresh()
+
+    def _get_shape(self):
+
+        return np.max(self.coords, axis=0) + 1
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+
+            self._set_view_slice(self.viewer.indices)
+
+        if self._need_visual_update:
+            self._need_visual_update = False
+            self._node.update()
+
+    def _refresh(self):
+        """Fully refresh the underlying visual.
+        """
+        self._need_display_update = True
+        self._update()
+
+    def _set_view_slice(self, indices):
+        """Sets the view given the indices to slice with.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
```

Comment:
So, by my reading of the code below, this only works when the displayed plane is the leading two dimensions, right?

Reply:
I think we'll have to revisit this when @kne42 has added the functionality to change which indices are being displayed as an image and which are being interacted with as sliders